autoload/jedi.vim

```diff
@@ -183,16 +183,27 @@ endif
 " ------------------------------------------------------------------------
 " functions that call python code
 " ------------------------------------------------------------------------
+function! jedi#goto()
+    PythonJedi jedi_vim.goto(mode="goto")
+endfunction
+
 function! jedi#goto_assignments()
-    PythonJedi jedi_vim.goto()
+    PythonJedi jedi_vim.goto(mode="assignment")
 endfunction
 
+let s:displayed_deprecation = 0
 function! jedi#goto_definitions()
-    PythonJedi jedi_vim.goto(is_definition=True)
+    if !s:displayed_deprecation
+        echohl WarningMsg
+        echom "Jedi-vim: jedi#goto_definitions is deprecated, please use jedi#goto instead."
+        echohl None
+        let s:displayed_deprecation = 1
+    endif
+    PythonJedi jedi_vim.goto(mode="definition")
```

Comment:
Can we not deprecate it? I think it's still useful for a few people (like me). I just don't want it to be there as a default, because 3 goto functions would definitely confuse people.